CRM/MembershipExtras/Hook/Pre/MembershipPaymentPlanProcessor/LineItem.php

```diff
@@ -16,52 +16,99 @@ public function __construct(&$params) {
    *
    * We here adjust the line total, unit price and tax amount
    * of the line item to be inline with the new contribution amount.
+   *
    */
   public function alterLineItemParameters() {
     if (isset($this->params['membership_type_id'])) {
-      $lineItemMembershipType = CRM_Member_BAO_MembershipType::findById($this->params['membership_type_id']);
-      //We only pro rate line item if price set is used and price field membership period type is fixed
-      if ($this->isUsingPriceSet() && $lineItemMembershipType->period_type == 'fixed') {
-        $this->calculateProRataLineItemAmounts($lineItemMembershipType, $this->membershipId);
-      }
-      else {
-        $this->calculateLineItemAmounts();
-      }
+      $this->handleMembershipTypeLineItem();
```

Comment:
After our last chat, I thought you will move the conditions to `calculateLineItemAmounts`, but this is better.